src/Compilers/Core/Portable/Syntax/ChildSyntaxList.cs

```diff
@@ -282,6 +320,15 @@ internal static SyntaxNodeOrToken ChildThatContainsPosition(SyntaxNode node, int
                 slotIndex++;
             }
 
+            if (slotIndex != slotData.SlotIndex)
+            {
+                slotData.SlotIndex = slotIndex;
+                slotData.PrecedingOccupantSlotCount = index - idx;
+
+                // Not used in this codepath
+                slotData.PositionAtSlotIndex = 0;
```

Comment:
Is there any chance this `SlotData` instance will be used subsequently in `ItemInternal()` which does use `PositionAtSlotIndex`?